src/server.c

```diff
@@ -133,23 +154,47 @@ void serverLogRaw(int level, const char *msg) {
     } else {
         int off;
         struct timeval tv;
-        int role_char;
         pid_t pid = getpid();
         int daylight_active = atomic_load_explicit(&server.daylight_active, memory_order_relaxed);
 
         gettimeofday(&tv, NULL);
         struct tm tm;
         nolocks_localtime(&tm, tv.tv_sec, server.timezone, daylight_active);
-        off = strftime(buf, sizeof(buf), "%d %b %Y %H:%M:%S.", &tm);
-        snprintf(buf + off, sizeof(buf) - off, "%03d", (int)tv.tv_usec / 1000);
+        switch (server.log_timestamp_format) {
+        case LOG_TIMESTAMP_LEGACY:
+            off = strftime(buf, sizeof(buf), "%d %b %Y %H:%M:%S.", &tm);
+            snprintf(buf + off, sizeof(buf) - off, "%03d", (int)tv.tv_usec / 1000);
+            break;
+
+        case LOG_TIMESTAMP_ISO8601:
+            off = strftime(buf, sizeof(buf), "%Y-%m-%dT%H:%M:%S.", &tm);
+            char tzbuf[7];
+            formatTimezone(tzbuf, sizeof(tzbuf), server.timezone, server.daylight_active);
+            snprintf(buf + off, sizeof(buf) - off, "%03d%s", (int)tv.tv_usec / 1000, tzbuf);
+            break;
+
+        case LOG_TIMESTAMP_MILLISECONDS:
+            snprintf(buf, sizeof(buf), "%ld.%03d", tv.tv_sec, (int)tv.tv_usec / 1000);
+            break;
+        }
```

Comment:
This is seconds with fractions. I believe what @madolson had in mind is just milliseconds since the epoch as an integer, as on this page: https://currentmillis.com/

@madolson Is that right?

Reply:
`tv.tv_sec * 1000 + tv.tv_usec`

It needs more than 32 bits. Note that `long` can be 32-bit on 32-bit platforms, so I think we need `long long` (`%lld`) for this.

Reply:
Code updated as you suggested.
```text
569917:C 1728914477183 * oO0OoO0OoO0Oo Valkey is starting oO0OoO0OoO0Oo
```

Reply:
Yes, the final result is what I expected.